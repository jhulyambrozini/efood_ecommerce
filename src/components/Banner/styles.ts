import styled from "styled-components"

export const Image = styled.div`
  background-repeat: no-repeat;
  background-size: cover;
  height: 17.5rem;
  width: 100%;
  display: block;
  position: relative;

  &::after {
    position: absolute;
    background-color: #000;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    content: '';
    opacity: 0.5;
  }
`

export const Titles = styled.div`
  z-index: 1;
  position: relative;
  display: grid;
  gap: 9.78rem;
  color: #FFF;
  font-size: 2rem;
  padding: 1.5rem 0 2rem 10.68rem;

  span {
    font-weight: 100;
  }

  h2 {
    font-weight: 900;
  }
`