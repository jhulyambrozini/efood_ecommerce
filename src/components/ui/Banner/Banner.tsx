import Image from '../Image'
import { BannerContainer, Titles } from './styles'

type BannerProps = {
  image: string
  type: string
  title: string
}

const Banner = ({ image, type, title }: BannerProps) => {
  return (
    <BannerContainer>
      <Image src={image} alt={title} />
      <Titles>
        <span>{type}</span>
        <h2>{title}</h2>
      </Titles>
    </BannerContainer>
  )
}

export default Banner
