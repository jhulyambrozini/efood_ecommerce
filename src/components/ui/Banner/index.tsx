import Image from '../../../utils/Image'
import { BannerContainer, Titles } from './styles'

export type BannerProps = {
  image: string
  type: string
  title: string
}

const Banner = ({ image, type, title }: BannerProps) => {
  return (
    <BannerContainer>
      <Image src={image} alt={title} />
      <Titles>
        <span>{type}</span>
        <h2>{title}</h2>
      </Titles>
    </BannerContainer>
  )
}

export default Banner
