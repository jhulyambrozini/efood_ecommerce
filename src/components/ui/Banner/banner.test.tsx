import { render, screen } from '@testing-library/react'
import Banner from '.'

const props = {
  image: 'img/img1.png',
  title: 'restaurante',
  type: 'italiana'
}

describe('<Banner />', () => {
  it('should be render correctly', () => {
    render(<Banner {...props} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.image
    )
    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()
    expect(screen.getByText(props.type)).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<Banner {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
