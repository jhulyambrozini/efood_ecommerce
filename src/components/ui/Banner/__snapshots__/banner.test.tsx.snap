// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should match snapshot 1`] = `
<div
  class="sc-dcJsrY gzPucS"
>
  <img
    alt="restaurante"
    class="loading"
    height="100"
    loading="lazy"
    src="img/img1.png"
    width="100"
  />
  <div
    class="sc-iGgWBj gmTjTR"
  >
    <span>
      italiana
    </span>
    <h2>
      restaurante
    </h2>
  </div>
</div>
`;
